File: frontend/src/domain/payment-category/repository.ts
import type { PaymentCategory } from "./entity";

/**
 * 指定されたuserIdに基づいて支出カテゴリーを取得
 * @param userId - ユーザーID
 * @returns 支出カテゴリーの配列
 */
export type FindPaymentCategoriesByUserId = (
	userId: string,
) => Promise<PaymentCategory[]>;

File: backend/domain/paymentcategory/repository.go
package paymentcategory

import (
	"context"
)

type Repository interface {
	// FindPaymentCategoriesByUserId : 指定されたuserIdに基づいて支出カテゴリーを取得
	// userId: ユーザーID
	// return: 支出カテゴリーの配列
	FindPaymentCategoriesByUserId(ctx context.Context, userId string) ([]PaymentCategory, error)
}
